File: source/includes/fts-tutorial/facet/tutorial-variable.js
const { MongoClient } = require("mongodb");

// connect to your Atlas cluster
const uri = "<connection-string>";
    
const client = new MongoClient(uri);

async function run() {
    try {
        await client.connect();

        // set namespace
        const database = client.db("sample_mflix");
        const coll = database.collection("movies");

        // define pipeline
        const agg = [
            {
              '$search': {
                'index': "facet-tutorial",
                'facet': {
                  'operator': {
                    'near': {
                      'path': 'released', 
                      'origin': new Date('Tue, 01 Nov 1921 00:00:00 GMT'), 
                      'pivot': 7776000000
                    }
                  }, 
                  'facets': {
                    'genresFacet': {
                      'type': 'string', 
                      'path': 'genres'
                    }, 
                    'yearFacet': {
                      'type': 'number', 
                      'path': 'year', 
                      'boundaries': [1910, 1920, 1930, 1940]
                    }
                  }
                }
              }
            }, {
              '$facet': {
                'meta': [
                  {
                    '$replaceWith': '$$SEARCH_META'
                  }, {
                    '$limit': 1
                  }
                ]
              }
            }, {
              '$set': {
                'meta': {
                  '$arrayElemAt': ['$meta', 0]
                }
              }
            }
        ];
        // run pipeline
        const result = coll.aggregate(agg);

        // print results
        await result.forEach((doc) => console.dir(JSON.stringify(doc)));
    } finally {
        await client.close();
    }
}
run().catch(console.dir);

File: source/includes/fts-tutorial/facet/tutorial-variable.go
package main

import (
	"context"
	"fmt"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {
	// connect to your Atlas cluster
	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI("<connection-string>"))
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(context.TODO())

	// set namespace
	collection := client.Database("sample_mflix").Collection("movies")

	// define pipeline stages
	searchStage := bson.D{{"$search", bson.M{
		"index": "facet-tutorial",
		"facet": bson.M{
			"operator": bson.M{
				"near": bson.M{
					"path":   "released",
					"origin": time.Date(1921, time.November, 1, 0, 0, 0, 0, time.UTC),
					"pivot":  7776000000},
			},
			"facets": bson.M{
				"genresFacet": bson.M{
					"path": "genres",
					"type": "string"},
				"yearFacet": bson.M{
					"path":       "year",
					"type":       "number",
					"boundaries": bson.A{1910, 1920, 1930, 1940}},
			},
		},
	}}}

	facetStage:= bson.D{{"$facet", bson.D{
        {"meta", bson.A{
            bson.D{{"$replaceWith", "$$SEARCH_META"}},
            bson.D{{"$limit", 1}},
        }},
    }}}
	setStage:= bson.D{{"$set", bson.D{
        {"meta", bson.D{
            {"$arrayElemAt", bson.A{"$meta",  0}},
        }},
    }}}

	// run pipeline
	cursor, err := collection.Aggregate(context.TODO(), mongo.Pipeline{searchStage, facetStage, setStage})
	if err != nil {
		panic(err)
	}

	// print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}
